### ISSUE TITLE
The estimate time and process seems not right

### ISSUE THREAD

**Issue by shuijing198799** [pingcap/tidb-lightning#431]
## Bug Report

Please answer these questions before submitting your issue. Thanks!

1. What did you do? If possible, provide a recipe for reproducing the error.
 Two estimate errors;

- Restart the tidb-lightning， the restore estimate time is not correct, it shows need 4h 15 min to  complete a file and after 5 min, whole task is completed.

- The file is complete 104% in the restore process

2. What did you expect to see?
show correct

3. What did you see instead?
none

4. Versions of the cluster

    - TiDB-Lightning version (run `tidb-lightning -V`):

        ```
       Release Version: v4.0.7-11-g2660ffe
       Git Commit Hash: 2660ffe03e85e2c84bce7a5dd4013360fd0f8e57
       Git Branch: master
       UTC Build Time: 2020-10-21 09:53:58
       Go Version: go version go1.13.12 linux/amd64
        ```

5. Operation logs
   - Please upload `tidb-lightning.log` for TiDB-Lightning if possible

[2020/10/27 09:37:33.298 +00:00] [INFO] [restore.go:601] [progress] **[files="4193/4033 (104.0%)"]** [tables="6/10 (60.0%)"] [speed(MiB/s)=90.41088115757076] [state=post-processing] []

[2020/10/26 13:24:35.689 +00:00] [INFO] [restore.go:601] [progress] [files="1936/4033 (48.0%)"] [tables="8/10 (80.0%)"] [speed(MiB/s)=34.929119006710025] [state=writing] **[remaining=4h14m33s]**
[2020/10/26 13:29:25.509 +00:00] [INFO] [restore.go:1639] ["analyze completed"] [table=`test`.`order_line`] [takeTime=1h7m41.459916195s] [][2020/10/26 13:29:25.510 +00:00] [INFO] [restore.go:698] ["restore table completed"] [table=`test`.`order_line`] [takeTime=3h59m49.814007382s] []
[2020/10/26 13:29:25.510 +00:00] [INFO] [restore.go:815] ["restore all tables data completed"] [takeTime=3h59m49.820567514s] []
[2020/10/26 13:29:25.510 +00:00] [INFO] [restore.go:1249] ["skip full compaction"][2020/10/26 13:29:25.510 +00:00] [INFO] [restore.go:570] ["everything imported, stopping periodic actions"]
[2020/10/26 13:29:26.096 +00:00] [INFO] [restore.go:1343] ["clean checkpoints start"] [keepAfterSuccess=false] [taskID=1603704574303293735]
[2020/10/26 13:29:26.096 +00:00] [INFO] [restore.go:1350] ["clean checkpoints completed"] [keepAfterSuccess=false] [taskID=1603704574303293735] [takeTime=173.607µs] []
[2020/10/26 13:29:26.096 +00:00] [INFO] [restore.go:312] ["the whole procedure completed"] **[takeTime=3h59m50.445529074s]** 
[][2020/10/26 13:29:26.097 +00:00] [INFO] [main.go:87] ["tidb lightning exit"]

--

6. Configuration of the cluster and the task
   - `tidb-lightning.toml` for TiDB-Lightning if possible
```
# lightning Configuration

[lightning]
check-requirements = true
index-concurrency = 4
level = "info"
file=""
pprof-port = 8289
table-concurrency = 4

[checkpoint]
enable = true
driver = "file"
dsn = "/etc/endpoint/cp.pb"

[security]
# Specifies certificates and keys for TLS connections within the cluster.
# Public certificate of the CA. Leave empty to disable TLS.
ca-path = "/var/lib/lightning-tls/ca.crt"
# Public certificate of this service.
cert-path = "/var/lib/lightning-tls/tls.crt"
# Private key of this service.
key-path = "/var/lib/lightning-tls/tls.key"

[tikv-importer]
backend = "local"
sorted-kv-dir = "/etc/endpoint"

[mydumper]
data-source-dir = "s3://tidblightning-cloud-test/data-14k?region=us-west-2"
no-schema = true
strict-format = true
#default-file-rules = false

#[[mydumper.files]]
#pattern = "(?i)^(?:[^/]*/)*([a-z]+)[0-9]*(?:\\.([0-9]+))?\\.(sql|csv)$"
#schema = "test"
#table = "$1"
#type = "$3"

[mydumper.csv]
backslash-escape = true
delimiter = '"'
#delimiter = ''
header = false
not-null = false
null = "NULL"
separator = ","

[tidb]
host = "db-tidb"
log-level = "error"
password = "12345678"
port = 4000
status-port = 10080
user = "root"
pd-addr = "db-pd:2379"
tls = "false"

[post-restore]
analyze = true
checksum = true

[cron]
log-progress = "5m"
switch-mode = "5m"
```

**Comment by glorv** [comment]
should be fix by #377

**Comment by shuijing198799** [comment]
```
[2020/11/12 09:09:25.455 +00:00] [INFO] [restore.go:625] [progress] [files="490/901 (54.4%)"] [tables="4/10 (40.0%)"] [chunks=490/901] [speed(MiB/s)=60.70536158544237] [state=writing] [remaining=4m12s]
[2020/11/12 09:14:25.458 +00:00] [INFO] [restore.go:625] [progress] [files="579/901 (64.3%)"] [tables="4/10 (40.0%)"] [chunks=579/901] [speed(MiB/s)=51.92817251929942] [state=writing] [remaining=5m34s]
[2020/11/12 09:19:25.504 +00:00] [INFO] [restore.go:625] [progress] [files="649/901 (72.0%)"] [tables="4/10 (40.0%)"] [chunks=649/901] [speed(MiB/s)=45.334801999784354] [state=writing] [remaining=5m49s]
[2020/11/12 09:24:25.465 +00:00] [INFO] [restore.go:625] [progress] [files="711/901 (78.9%)"] [tables="4/10 (40.0%)"] [chunks=711/901] [speed(MiB/s)=40.76245572479163] [state=writing] [remaining=5m21s]
[2020/11/12 09:29:25.466 +00:00] [INFO] [restore.go:625] [progress] [files="762/901 (84.6%)"] [tables="4/10 (40.0%)"] [chunks=762/901] [speed(MiB/s)=37.84552835500362] [state=writing] [remaining=4m34s]
[2020/11/12 09:34:25.449 +00:00] [INFO] [restore.go:625] [progress] [files="801/901 (88.9%)"] [tables="4/10 (40.0%)"] [chunks=801/901] [speed(MiB/s)=34.79081975192149] [state=writing] [remaining=3m45s]
[2020/11/12 09:39:25.454 +00:00] [INFO] [restore.go:625] [progress] [files="864/901 (95.9%)"] [tables="5/10 (50.0%)"] [chunks=864/901] [speed(MiB/s)=36.851693923751654] [state=writing] [remaining=1m30s]

```
It seems fixed
